torch/distributed/_shard/sharding_spec/chunk_sharding_spec.py

```diff
@@ -166,10 +167,15 @@ def shard(self, tensor: torch.Tensor, src_rank: int = 0, process_group=None) ->
         assert local_metadata is not None
 
         # Scatter the shards to all ranks in the pg
+        # scatter takes the global rank as ``src``
```

Comment:
is the scatter behavior correct? I think scatter need to take local rank as `src` if it passes a different pg than the global pg.

Reply:
I agree with you that scatter behavior doesn't seem right, but I'm not sure if we can change it without a BC breaking change.

Reply:
Also, it looks like `gather`, `broadcast`, `send`, `recv` all have the same behavior. So this might probably need some deeper investigation.

Reply:
Yeah make sense, let's go ahead and merge it, fixing c10d can be a separate discussion.